src/overlays/actors/ovl_Bg_Spot11_Oasis/z_bg_spot11_oasis.c

```diff
@@ -19,25 +25,136 @@ const ActorInit Bg_Spot11_Oasis_InitVars = {
     (ActorFunc)BgSpot11Oasis_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot11_Oasis/func_808B27F0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot11_Oasis/func_808B280C.s")
+s16 D_808B2E10[][2] = {
+    { 1260, 2040 }, { 1259, 1947 }, { 1135, 1860 }, { 1087, 1912 }, { 1173, 2044 },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+Vec3f D_808B2E34[] = {
+    { 0.0f, -100.0f, 0.0f },   { 100.0f, -80.0f, -50.0f }, { -50.0f, -80.0f, -100.0f },
+    { -75.0f, -90.0f, 90.0f }, { 30.0f, -100.0f, 40.0f },
+};
+
+extern Gfx D_06000870[];
+
+void func_808B27F0(GlobalContext* globalCtx, s16 arg1) {
+    WaterBox* waterBoxes;
+    waterBoxes = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    waterBoxes->unk_02 = arg1;
```

Comment:
Since `waterBoxes` is a pointer to an array of water boxes, I think this is specifically taking the first water box in the array and setting a value inside it. So I suggest adjusting the code like this:
```suggestion
    WaterBox* waterBox = &globalCtx->colCtx.stat.colHeader->waterBoxes[0];

    waterBox->unk_02 = arg1;
```